src/rpc/rawtransaction.cpp

```diff
@@ -246,14 +246,15 @@ static RPCHelpMan gettxoutproof()
     std::set<uint256> setTxids;
     uint256 oneTxid;
     UniValue txids = request.params[0].get_array();
+    if (txids.empty()) {
```

Comment:
you can move this to after the for loop and run the check on setTxids, then remove oneTxid and replace it with setTxids.front()

Reply:
ATM `oneTxid` is the first `txid` in the array, with you suggestion that wouldn't be the case since `set::set` orders, right?

Reply:
oneTxid can be any txid

Reply:
Right, all must be in same block.

Reply:
Done, thanks!